### ISSUE TITLE
MongoDB: support setting pod and statefulset labels

### ISSUE THREAD

**Issue by sfitts** [kubedb/project#544]
Based on both the documentation and some experimentation it seems that you cannot currently set the labels for either the stateful set or pods created by KubeDB.  Pod labeling is very important to us so this is a major pain point.

**Comment by the-redback** [comment]
statefulset labels can be set like this.
```
apiVersion: kubedb.com/v1alpha1
kind: MongoDB
metadata:
  name: mongo-sh
  namespace: demo
  labels:
    mongodb: mg
spec:
  version: 3.6-v4
  shardTopology:
    configServer:
      replicas: 3
      storage:
        resources:
          requests:
            storage: 1Gi
        storageClassName: standard
    mongos:
      replicas: 2
      strategy:
        type: RollingUpdate
    shard:
      replicas: 3
      shards: 3
      storage:
        resources:
          requests:
            storage: 1Gi
        storageClassName: standard
  updateStrategy:
    type: RollingUpdate
  storageType: Durable
  terminationPolicy: WipeOut
```

Here, label `mongodb: mg` is forwarded to statefulsets.

**Comment by sfitts** [comment]
Interesting... is that documented somewhere?  

I guess I assumed you had to put them in the `spec.podTemplate.controller` section since that's where the annotations go (and `spec.podTemplate.metadata` for pods).  But any labels placed there are ignored.

Is there a way to set pod labels?

**Comment by BufferUnderflower** [comment]
Also searching a way to label pods via crd (Istio needs app&version labels). @the-redback is it possible?

**Comment by garfieldcoked** [comment]
@BufferUnderflower, a bit new to k8s, istio (and even newer to kubedb). Why does Istio need app and version label? After seeing your comment I  reviewed the docs and this was recommended https://istio.io/docs/setup/additional-setup/requirements/. Is there something else this is used for?

I am currently having issues (with Postgres) and I am wondering if it's related to labelling  
```
Envoy proxy is NOT ready: 2 errors occurred:
...
envoy missing listener for inbound application port: 5432
```

**Comment by AviMoto** [comment]
Hi The labels are important for metrics and monitor, tool like kiali will visualize you mesh and grouped the kubedb clusters as an app, In addition it will help to add "<protocol>-" before the port name in the  clusters services ([istio pod and services](https://istio.io/docs/setup/additional-setup/requirements/))

**Comment by thst71** [comment]
This is also required for Postgres, the same issue exists there. Annotations are taken over, but pod labels are explicitly set. 
I suggest, that all podTemplate.metadata.labels are included that do not violate the kubed namespace and are not already in OffshootSelector() (from postgres operator, may vary for mongodb)

**Comment by tal-ayalon** [comment]
Same problem I have in MySQL. I can't label pods

**Comment by tamalsaha** [comment]
Fixed in master. Will be available in upcoming v2021.11.18 release.

**Comment by tamalsaha** [comment]
Please note, this support has been added in v2021.11.24 release. You can find the release notes here: https://blog.byte.builders/post/kubedb.v2021.11.24/
